lib/tasks/data.rake

```diff
@@ -75,4 +76,137 @@ namespace :aleftng do
       end
     end
   end
+
+  def find_and_replace_in_text(find, replace, text)
+    #puts "Search and replace #{find.inspect} => #{replace.inspect}"
+    text.gsub!(find, replace)
+    return text
+  end
+
+  # TODO - Výnimka - iný typ otázky ako tieto základné
+  def find_question_type(text)
+    if text == "single-choice"
+      return "SingleChoiceQuestion"
+    elsif text == "multi-choice"
+      return "MultiChoiceQuestion"
+    elsif text == "answer-validator"
+      return "EvaluatorQuestion"
+    elsif text == "complement"
+      return "Complement"
+    end
+  end
+
+  def find_substring(text)
+    if text.include? "<correct>"
+      return true
+    else
+      return false
+    end
+  end
+
+  def import_Choice_questions(dir)
+    # Prečitanie súboru a vynechanie vypísania hlavičky pri každom zázname
+    parsed_file = CSV.read(dir, :headers => false)
+    parsed_file.each do |row|
+
+      # Predspracovanie
+      zero_version = row[4]
+      picture = row[1]
+      external_reference = row[0]
+      question_text = PandocRuby.new(row[5], :from => :docbook, :to => :markdown)
+      question_text = find_and_replace_in_text("\n", "", question_text.to_s)
+      answers = row[11]
+      question_type = find_question_type(row[9])
+
+      # Vyberieme otázky do nultej verzie a bez obrázku
+      if (!zero_version.nil? && picture.nil?)
+        lo = LearningObject.find_by_external_reference(external_reference)
+        if (lo.nil?)
+          #puts "QUESTION NOT EXISTS"
+          lo = LearningObject.create!( type: question_type, question_text: question_text, external_reference: external_reference )
+          #puts "QUESTION: #{question_text}"
+          splitted_answers = find_and_replace_in_text(";", "\n", answers).split(/\r?\n/)
+          splitted_answers.each do |answer|
+            correct_answer = find_substring(answer)
+            answer_text = PandocRuby.new(answer, :from => :docbook, :to => :markdown)
+            answer_text = find_and_replace_in_text("\n", "", answer_text.to_s)
+            Answer.create!( learning_object_id: lo.id, answer_text: answer_text, is_correct: correct_answer )
+            #puts "ANSWER: #{answer} | #{answer_text} | #{correct_answer}"
+          end
+          #puts "QUESTION NOT EXISTS"
+        else
+          #puts "QUESTION EXISTS"
+          lo.update( type: question_type, question_text: question_text )
+          #puts "QUESTION: #{question_text}"
+          ans = Answer.where(learning_object_id: lo.id)
+          ans.destroy_all
```

Comment:
Vo vztahoch user_to_lo budu priebezne zalogovane ID odpovedi, ktore pouzivatel pri zodpovedani otazky vybral, ale importom uz existujucej otazky sa vsetky odpovedie zrusia a vytvoria nanovo (s inymi ID a z logov teda uz nebudeme vediet povedat, ktore odpovede boli v danej interakcii vybrane). Toto nie je dobre.

Rozmyslam, co s tym:
Jedna moznost je nechat tam predchadzajuce odpovede a zneplatnit ich (nastavit im nejaky atribut na false a potom ich v zobrazovani a vyhodnocovani odpovedi neuvazovat, idealne cez scope, napr. spravit ```lo.available_answers```, ktore vrati len platne odpovede). Ale zbytocne si to komplikujeme a zase robit toto (zneplatnenie starych odpovedi a vytvorenie novych) pri kazdom importe, ked sa odpovede mozno ani nezmenili je tiez zbytocne.

Spravme to zatial tak, ze odpovede sa pri importe uz existujucej otazky nemenia, t.j. zostanu v DB tie, co tam su. (Podla info by mal byt text a odpovede v CSV zatial stabilne a neskor planujeme umoznit zmeny cez editacne rozhranie, opakovany import potrebujeme najma kvoli konceptom a dalsim metadatam.)